### DISCUSSION TITLE
How to update layout outside of {children} for pages?

### DISCUSSION THREAD

**Discussion by wamiswag** [vercel/next.js#54883] [App Router]
**The need**

I have a layout defined - which is mostly shared amongst all of the pages. But, for the homepage only, I need to add some specific HTML code inside of the <header> element. I do not want to restructure my HTML code - because it is intended to mimic our production HTML code. I also want to avoid creating separate layouts for the homepage, and all other pages (using route groups) because it goes against DRY (don't repeat yourself).

I also do not want to implement something that requires making it a client component. I should be able to render a static banner on my homepage within a shared HTML element without forcing it to run client-side...

**The ask**

Does anyone have a good solution for this - which follows Next JS standards - but also prevents adding in superfluous code? I've read a bit on parallel routes, and that seems... interesting.

This is a very common use case (there are already a few Stack Overflow pages with users asking the same question, but having no good direction)

**Comment by avisra** [comment]
Agreed. This a big need.

**Comment by devjiwonchoi** [comment]
Using Route Groups, set your main `page.tsx` inside the group somewhat like `(home)`, and add codes to the layout inside the group.

![Screenshot 2023-09-01 at 10 51 03 PM](https://github.com/vercel/next.js/assets/120007119/6a78a30a-e91e-413c-8951-a798de324699)

Note that you shouldn't have `page.tsx` on the root segment, `app/page.tsx`.

**Reply by avisra** [reply]
Yea - as mentioned in the description - we'd like to avoid the route groups approach if possible, as it leads to unnecessarily duplicated code

**Reply by devjiwonchoi** [reply]
I see, my bad. Don't think there's an alternate way to resolve this for now.
